packages/assets/src/loader/parsers/textures/utils/createTexture.ts

```diff
@@ -10,16 +10,25 @@ export function createTexture(base: BaseTexture, loader: Loader, url: string)
     base.resource.internal = true;
 
     const texture = new Texture(base);
-
-    // remove the promise from the loader and the url from the cache when the texture is destroyed
-    texture.baseTexture.once('destroyed', () =>
+    const unload = () =>
     {
         delete loader.promiseCache[url];
 
         if (Cache.has(url))
         {
             Cache.remove(url);
         }
+    };
+
+    // remove the promise from the loader and the url from the cache when the texture is destroyed
+    texture.baseTexture.once('destroyed', unload);
```

Comment:
```suggestion
   base.once('destroyed', unload);
```

Reply:
If the base texture is destroyed, we surely want to remove it from the asset manager, no? Imagine someone destroys only the base texture, but not the texture. We probably don't want the texture with a destroyed base texture to remain in the asset manager.

Maybe we should have a warning to the base texture case as well?

Reply:
We will have similar problems with bitmap fonts, spritesheets, etc. which generally all have a destroy path too. I think we need a more wholistic approach to this problem: encouraging unload over destroy.